### PR TITLE
Follow up https://github.com/rubocop/rubocop-rspec/pull/1768#discussion_r1442928428

### PR DESCRIPTION
**Pull request by ydah**
https://github.com/rubocop/rubocop-rspec/pull/1768#discussion_r1442928428

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [-] Added tests.
- [-] Updated documentation.
- [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: lib/rubocop/cop/rspec/shared_examples.rb [hunk 1]
```diff
@@ -46,14 +46,14 @@ class SharedExamples < Base
         # @!method shared_examples(node)
         def_node_matcher :shared_examples, <<~PATTERN
           {
-            (send #rspec? #SharedGroups.all ...)
-            (send nil? #Includes.all ...)
+            (send #rspec? #SharedGroups.all _+)
+            (send nil? #Includes.all _+)
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/shared_examples.rb:50]: If you instead do this
```suggestion
            (send #rspec? #SharedGroups.all $_ ...)
            (send nil? #Includes.all $_ ...)
```
then `ast_node` will be a block argument in line 55.

File: lib/rubocop/cop/rspec/shared_examples.rb [hunk 2]
```diff
@@ -46,14 +46,14 @@ class SharedExamples < Base
...
           {
-            (send #rspec? #SharedGroups.all ...)
-            (send nil? #Includes.all ...)
+            (send #rspec? #SharedGroups.all $_ ...)
+            (send nil? #Includes.all $_ ...)
           }
         PATTERN
 
         def on_send(node)
           shared_examples(node) do
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/shared_examples.rb:55]:
```suggestion
          shared_examples(node) do |ast_node|
```

File: lib/rubocop/cop/rspec/shared_examples.rb [hunk 3]
```diff
@@ -46,14 +46,14 @@ class SharedExamples < Base
...
-            next unless (ast_node = node.first_argument)
+            ast_node = node.first_argument
```
Comment (bquorning) [inline lib/rubocop/cop/rspec/shared_examples.rb:56]:
```suggestion
```

**Review by bquorning — APPROVED** [review]
🎉

**Review by Darhazer — APPROVED** [review]

**Pull request merged**